Watchman.Discord/Areas/UselessFeatures/Controllers/UselessController.cs

```diff
@@ -21,5 +37,33 @@ public async Task PrintMarchew(MarchewCommand command, Contexts contexts)
             var messagesService = _messagesServiceFactory.Create(contexts);
             await messagesService.SendMessage(text);
         }
+
+        // do testu obsługi reackji
+        // ta komenda z parametrem "on" aktywuje możliwość przyznania roli python i usunięcia tej roli gdy zareaguje się na konkretną wiadomość (w sensie o konkretnej treści)
+        [DiscordCommand("MessageRoleOnRequest")]
+        public void AddRoleOnRequest(DiscordRequest request, Contexts contexts)
+        {
+            if (request.Arguments?.FirstOrDefault()?.Name == "on")
+            {
+                _reactionsService.AddOnUserAddedReaction(rc => 
+                {
+                    if (rc.MessageContext.Content.Trim() == "Like = dostajesz role python" && rc.EmoteName == "👍") 
```

Comment:
nie chcemy uzależniać sie od treści wiadomości
i nie chcemy żeby to obsługiwało wiadomość od dowolnego użytkownika 
to spowoduje więcej problemów niż korzyści 

powinniśmy reagować na wiadomość bota
przykładowo użytkownik wpisuje 

`-CreateReactionMessage -AddRole python -OnReaction :python: -AddRole csharp -OnReaction :csharp:`
(zakładając że reakcje :python: i :csharp: istnieją)

Wtedy generuje wiadomość
```
Dodaj reakcje :python: aby dostać sobie role python
Dodaj reakcje :csharp: aby dodać sobie role csharp
``` 
(discord powinien wyświetlić ikonke reakcji w miejscu :tekst:)
Po czym dodaje te reakcje do swojej wiadomości, żeby użytkownik wiedział co kliknąć

Zapisuje w bazie ID wiadomości której ma nasłuchiwać 
I jak ktoś doda reakcje do wiadomości o ID którego nasłuchujemy, to sprawdzamy czy dodał reakcje której nasłuchujemy
Jeśli wiadomość i reakcja się zgadzają, to przekazujemy do metody odpowiednie parametry 
-kto dodał
-do której wiadomości
-jaką reakcje 
(czyli to co zbierasz w ReactionContext)

Reply:
Przypominam, że to tylko przykład, dzięki któremu jest możliwe sprawdzenie czy w wygodny i intuicyjny sposób można korzystać z obsługi reakcji. Ja nie chciałem wymyślać czegoś co byłoby "perfekcyjne" i zarazem byłoby trzeba siedzieć nad tym sporo czasu aby to dobrze zrobić, to po prostu zwykły przykład. W tej chwili nie jest najważniejsza logika tego przykładu.

Więc dobrze się sprawdza ten mechanizm czy nie? Moim zdaniem tak, ale można by ewetualnie coś zmienić w ReactionContext aby móc łatwiej uzyskać dostęp do poszczególnych danych zamiast się "dokopywać" do nich.

PS. Rozumiem, że mogę już usunąć ten przykład? Już nie jest potrzebny?

Reply:
`Ja nie chciałem wymyślać czegoś co byłoby "perfekcyjne" i zarazem byłoby trzeba siedzieć nad tym sporo czasu aby to dobrze zrobić`

`Więc dobrze się sprawdza ten mechanizm czy nie?`

Właśnie na tym polega problem - że na razie to nie jest kompletne 
Na razie to nie jest system gotowy do użycia

Dlatego zaproponowałem żeby ktoś z nas to dokończył, a ty byś dostał inne zadanie

Reply:
proszę cię... nie wyrywaj moich słów z kontekstu. Te słowa się odnoszą do przykładu a nie do obsługi reakcji .

Sama obsługa reakcji jest prawie skończona.

Reply:
No to zrób pełny przykład, a nie testowy, który pokazuje praktyczne zastosowanie tego systemu

Na pewno zajmie to mniej niż 20 poprawek 
Bo ja od razu widze że poza przykładami testowymi nie da sie tego wykorzystać

Reply:
"Bo ja od razu widze że poza przykładami testowymi nie da sie tego wykorzystać" skąd ta śmiała teza? Jak się nie da? Dostęp do ReactionService masz wszędzie bo jest w IoC, większość danych jest w ReactionContext a inne mogą być przechwytane przez lambdy dołączone do delegatu ReactionService. 

"No to zrób pełny przykład, a nie testowy, który pokazuje praktyczne zastosowanie tego systemu" - chciałeś aby po zalajkowaniu dodało role, więc to właśnie zrobiłem, to nie ma znaczenia czy przykład ma inne duperele niezwiązane z reakcjami czy nie, bo ja dałem tam wszystko co jest związane z tym przypadkiem.

Poza tym miałem się zająć reakcjami, a nie wymyślaniem jakiś innych dziwnych rzeczy, jeszcze w dodatku robię cały czas zgodnie z waszymi zaleceniami i ciągle jest coś nie tak.

"Na pewno zajmie to mniej niż 20 poprawek" - sam mówiłeś, że rzekomo "zadanie stoi w miejscu" i jeszcze w dodatku chcesz je wydłużyć o rzeczy (czyli dodawanie roli) nie związane z obsługą reakcji.

Reply:
> Poza tym miałem się zająć reakcjami, a nie wymyślaniem jakiś innych dziwnych rzeczy, jeszcze w dodatku robię cały czas zgodnie z waszymi zaleceniami i ciągle jest coś nie tak.
powiem tak, spójrz na mój ostatni PR
miałem tylko dokończyć ten task, który miał +200 linii
ostatecznie po jakoś dwóch tygodniach, skończyłem z +1200 linii bo zrobiłem masę pobocznych rzeczy
trudno żeby na każdą drobnostkę był oddzielny task w trello
dlatego bardzo dużo robimy przy okazji

Reply:
```
"Bo ja od razu widze że poza przykładami testowymi nie da sie tego wykorzystać" skąd ta śmiała teza? Jak się nie da? Dostęp do ReactionService masz wszędzie bo jest w IoC, większość danych jest w ReactionContext a inne mogą być przechwytane przez lambdy dołączone do delegatu ReactionService.
```

Dostęp i możliwość wykorzystania to dwie różne rzeczy 

```
"No to zrób pełny przykład, a nie testowy, który pokazuje praktyczne zastosowanie tego systemu" - chciałeś aby po zalajkowaniu dodało role, więc to właśnie zrobiłem, to nie ma znaczenia czy przykład ma inne duperele niezwiązane z reakcjami czy nie, bo ja dałem tam wszystko co jest związane z tym przypadkiem.
```
Nie zrobiłeś bo nie zadziała obsługa reakcji gdyby bot sie zrestartował
Miałeś zrobić kompletny przykład, a nie testowy który pokazuje że dostajemy informacje o reakcjach 

```
Poza tym miałem się zająć reakcjami, a nie wymyślaniem jakiś innych dziwnych rzeczy, jeszcze w dodatku robię cały czas zgodnie z waszymi zaleceniami i ciągle jest coś nie tak.
```
To samo co napisał @gagyn 
Celem nie jest zaliczenie zadania (tak jak w szkole) tylko zrobienie czegoś co działa dobrze 
W pracy też jakoś nie zauważyłem żebym miał osobne zadanie do wszystkiego, zazwyczaj to jest kilka zdań napisanych na szybko, a później dorzuca sie dodatkowe rzeczy bo wynika to z sytuacji (coś sie zmieniło w trakcie)
Celem programisty jest sprawianie żeby ostateczny użytkownik miał to czego potrzebuje, a nie zaliczanie zadań
Zadania to tylko krótkie notatki napisane w jakimś tam czasie żeby opisać w skrócie o co chodzi, a reszte już sie ustala prywatnie (w 90% przypadków)

```
"Na pewno zajmie to mniej niż 20 poprawek" - sam mówiłeś, że rzekomo "zadanie stoi w miejscu" i jeszcze w dodatku chcesz je wydłużyć o rzeczy (czyli dodawanie roli) nie związane z obsługą reakcji.
```
Chodziło o to że zadanie się bardzo wydłuża, a progres jest bardzo mały
Przebudowa połowy mechanizmu w 1 wieczór to nie to samo co drobne zmiany, gdzie po każdej jest potrzebne review i tłumaczenie... co ostatecznie zabiera dużo czasu 3 osobom